File: php/testcode/benchmark_test_00835.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00835(BenchmarkRequest $req): BenchmarkResponse {
    $key = random_bytes(32);
    $iv = random_bytes(16);
    $enc = openssl_encrypt($req->param('data'), 'AES-256-CBC', $key, 0, $iv);
    return BenchmarkResponse::ok($enc);
}

File: rust/testcode/benchmark_test_00835.rs
use regex::Regex;

fn safe_pattern(_user: &str) -> &'static str {
    r"^[a-z]+$"
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let user_input = req.param("pattern");
    let text = req.param("text");

    let re = match Regex::new(safe_pattern(&user_input)) {
        Ok(r) => r,
        Err(e) => return super::shared::BenchmarkResponse::bad_request(&e.to_string()),
    };

    let found = re.is_match(&text);
    super::shared::BenchmarkResponse::ok(&format!("Match: {}", found))
}
